File: Baas.php/takeout_admin/Statistics.php
<?php

namespace takeout_admin;

class Statistics 
{
    
	public static function getCuisineStatistics($params, $context) 
    {
		$startDate = isset($_REQUEST["startDate"]) ? $_REQUEST["startDate"] : null;
		$endDate = isset($_REQUEST["endDate"]) ? $_REQUEST["endDate"] : null;
		
		$sqlParams = array();
        array_push($sqlParams, $startDate);
		array_push($sqlParams, $endDate);
		//$startDate = '2016-03-04 16:44:47';
        //$endDate = '2014-03-04 16:44:47';
		
		$table = null;
		$conn = $context->Connection("demo");
		$sql = "";
		try{
			if($startDate != null && $endDate != null){
				$sql = "SELECT SUM(ord.fSum) AS daySum ,count(*) AS count,avg(ord.fSum) AS price,date(fCreateTime) AS times FROM takeout_order ORD where ord.fCreateTime between ? and ? GROUP BY date(ord.fCreateTime)";
				$table = \baas\data\Util::queryDataBySQL($conn, $sql, $sqlParams, null, null, null);
			}else{
				$sql = "SELECT SUM(ord.fSum) AS daySum ,count(*) AS count,avg(ord.fSum) AS price,date(fCreateTime) AS times FROM takeout_order ORD GROUP BY date(ord.fCreateTime)";
				$table = \baas\data\Util::queryDataBySQL($conn, $sql, null, null, null, null);                
			}			
			
			$result = \baas\data\Transform::tableToJson($table);
            echo json_encode($result);
			
		}
        catch(Exception $e)
        {
            echo $e->getMessage();
        }
	}
}

?>

File: Baas/takeoutAdmin/Statistics.java
package takeoutAdmin;

import java.sql.Connection;
import java.util.ArrayList;
import java.util.List;

import javax.naming.NamingException;

import com.alibaba.fastjson.JSONObject;
import com.justep.baas.action.ActionContext;
import com.justep.baas.data.DataUtils;
import com.justep.baas.data.Table;
import com.justep.baas.data.Transform;
import com.justep.baas.data.sql.SQLException;

public class Statistics {
	public static JSONObject getCuisineStatistics(JSONObject params, ActionContext context) throws SQLException, NamingException, java.sql.SQLException {
		
		String startDate = params.getString("startDate");
		String endDate = params.getString("endDate");
		
		List<Object> sqlParams = new ArrayList<Object>();
		sqlParams.add(startDate);
		sqlParams.add(endDate);
		
		
		Table table = null;
		Connection conn = context.getConnection("demo");
		String sql = "";
		try{
			if(startDate != null && endDate != null){
				sql = "SELECT SUM(ord.fSum) AS daySum ,count(*) AS count,avg(ord.fSum) AS price,date(fCreateTime) AS times FROM takeout_order ORD where ord.fCreateTime between ? and ? GROUP BY date(ord.fCreateTime)";
				table = DataUtils.queryData(conn, sql, sqlParams, null, null, null);
			}else{
				sql = "SELECT SUM(ord.fSum) AS daySum ,count(*) AS count,avg(ord.fSum) AS price,date(fCreateTime) AS times FROM takeout_order ORD GROUP BY date(ord.fCreateTime)";
				table = DataUtils.queryData(conn, sql, null, null, null, null);

			}
			
			
			return Transform.tableToJson(table);
			
		} finally {
			conn.close();
		}
		
	
	}
}
